.github/workflows/build.yml

```diff
@@ -0,0 +1,98 @@
+name: Build
+
+on: [push]
+
+jobs:
+  Linux:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y build-essential cmake libgl-dev libxrandr-dev libxinerama-dev libxcursor-dev libxi-dev
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - name: Build
+        run: |
+          cmake -B bin -DCMAKE_BUILD_TYPE=Release .
+          cmake --build bin
+      - name: Publish artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: OpenNFS-Linux
+          path: |
+            bin/OpenNFS
+            resources
+            shaders
+            LICENSE
+            README.md
+
+  Windows:
+    runs-on: windows-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - name: Build
+        run: |
+          cmake -B bin -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE=C:\vcpkg\scripts\buildsystems\vcpkg.cmake -DVCPKG_TARGET_TRIPLET=x64-windows -A x64 .
```

Comment:
This should use ninja as well, but it is a single-configuration builder, so needs slight setup changes: https://github.com/openblack/openblack/pull/68

Reply:
I was not able to get this to work, so I am not able to tell you if this would be faster, but I did add `--parallel 4` to shave off 2 to 3 minutes.